### PR TITLE
makefile: remove install, go get use origin GOPATH

### PR DESCRIPTION
**Pull request by tiancaiamao**
`go get` use the origin GOPATH
`go build` use _vendor:GOPATH
`make install` is removed, which is seldom used. 
A problem is that  `go install` never used, and there isn't a GOPATH/pkg, all code is rebuild everytime.

fix issue https://github.com/pingcap/tidb/issues/1778

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -63,7 +61,7 @@ parser: goyacc
 
 check:
 	bash gitcookie.sh
-	go get github.com/golang/lint/golint
+	GOPATH=$(ORIGIN_GOPATH) go get github.com/golang/lint/golint
```
Comment (shenli) [inline Makefile]: Make an alias for "GOPATH=$(ORIGIN_GOPATH) go get"

File: Makefile [hunk 2]
```diff
@@ -121,7 +120,7 @@ benchkv:
 
 update:
 	which glide >/dev/null || curl https://glide.sh/get | sh
```
Comment (overvenus) [inline Makefile]: The script at `https://glide.sh/get` uses `GOPATH` too.

**Comment by tiancaiamao** [comment]
So, we just use GOPATH=_vendor:$GOPATH when build or test, and don't export GOPATH
Nothing else special, following the standard Go rules.
Is that ok? @overvenus

**Comment by overvenus** [comment]
Yes, it looks good to me. :+1:

**Comment by shenli** [comment]
LGTM

**Pull request merged**